src/main/java/org/tron/common/runtime/vm/PrecompiledContracts.java

```diff
@@ -90,6 +106,36 @@ public static PrecompiledContract getContractForAddress(DataWord address) {
         public abstract long getGasForData(byte[] data);
 
         public abstract Pair<Boolean, byte[]> execute(byte[] data);
+
+        private byte[] callerAddress;
+
+        public void setCallerAddress(byte[] callerAddress) {
+            this.callerAddress = callerAddress;
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Store a copy of "callerAddress". ([squid:S2384](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2384))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1172&issues=AWTG3tSAxXt_iLiSomzj&open=AWTG3tSAxXt_iLiSomzj)

[ISSUE_KEY]: AWTG3tSAxXt_iLiSomzj